Mathlib/CategoryTheory/Functor/Basic.lean

```diff
@@ -0,0 +1,140 @@
+
+/-- The prefunctor between the underlying quivers. -/
+add_decl_doc Functor.toPrefunctor
+
+end
+
+/-- Notation for a functor between categories. -/
+-- A functor is basically a function, so give ⥤ a similar precedence to → (25).
+-- For example, `C × D ⥤ E` should parse as `(C × D) ⥤ E` not `C × (D ⥤ E)`.
+infixr:26 " ⥤ " => Functor -- type as \func
+
+attribute [simp] Functor.map_id
+attribute [reassoc, simp] Functor.map_comp
+
+namespace Functor
+
+section
+
+variable (C : Type u₁) [Category.{v₁} C]
+
+initialize_simps_projections Functor (toPrefunctor_obj → obj,
+  toPrefunctor_map → map, -toPrefunctor)
+
+-- We don't use `@[simps]` here because we want `C` implicit for the simp lemmas.
+/-- `𝟭 C` is the identity functor on a category `C`. -/
+protected def id : C ⥤ C where
+  obj X := X
+  map f := f
+#align category_theory.functor.id CategoryTheory.Functor.id
+
+/-- Notation for the identity functor on a category. -/
+notation "𝟭" => Functor.id -- Type this as `\sb1`
+
+instance : Inhabited (C ⥤ C) :=
+  ⟨Functor.id C⟩
+
+variable {C}
+
+@[simp]
+theorem id_obj (X : C) : (𝟭 C).obj X = X := rfl
+#align category_theory.functor.id_obj CategoryTheory.Functor.id_obj
+
+@[simp]
+theorem id_map {X Y : C} (f : X ⟶ Y) : (𝟭 C).map f = f := rfl
+#align category_theory.functor.id_map CategoryTheory.Functor.id_map
+
+end
+
+section
+
+variable {C : Type u₁} [Category.{v₁} C] {D : Type u₂} [Category.{v₂} D]
+  {E : Type u₃} [Category.{v₃} E]
+
+/-- `F ⋙ G` is the composition of a functor `F` and a functor `G` (`F` first, then `G`).
+-/
+@[simps obj]
+def comp (F : C ⥤ D) (G : D ⥤ E) : C ⥤ E where
+  obj X := G.obj (F.obj X)
+  map f := G.map (F.map f)
+#align category_theory.functor.comp CategoryTheory.Functor.comp
+
+/-- Notation for composition of functors. -/
+infixr:80 " ⋙ " => comp
+
+@[simp]
+theorem comp_map (F : C ⥤ D) (G : D ⥤ E) {X Y : C} (f : X ⟶ Y) :
+  (F ⋙ G).map f = G.map (F.map f) := rfl
+#align category_theory.functor.comp_map CategoryTheory.Functor.comp_map
+
+-- These are not simp lemmas because rewriting along equalities between functors
+-- is not necessarily a good idea.
+-- Natural isomorphisms are also provided in `whiskering.lean`.
+protected theorem comp_id (F : C ⥤ D) : F ⋙ 𝟭 D = F := by cases F; rfl
+#align category_theory.functor.comp_id CategoryTheory.Functor.comp_id
+
+protected theorem id_comp (F : C ⥤ D) : 𝟭 C ⋙ F = F := by cases F; rfl
+#align category_theory.functor.id_comp CategoryTheory.Functor.id_comp
+
+@[simp]
+theorem map_dite (F : C ⥤ D) {X Y : C} {P : Prop} [Decidable P]
+    (f : P → (X ⟶ Y)) (g : ¬P → (X ⟶ Y)) :
+    F.map (if h : P then f h else g h) = if h : P then F.map (f h) else F.map (g h) := by aesop_cat
+#align category_theory.functor.map_dite CategoryTheory.Functor.map_dite
+
+-- Porting note: `to_prefunctor_obj` and `to_prefunctor_map` are now tautologies,
+-- so have not been ported.
+
+@[simp]
+theorem to_prefunctor_comp (F : C ⥤ D) (G : D ⥤ E) :
```

Comment:
I am wondering whether `to_prefunctor_comp` should be named `toPrefunctor_comp`? If so, there are a few other names which should be changed.